pyclowder/connectors.py

```diff
@@ -374,19 +375,21 @@ def _process_message(self, body):
         if body.get('notifies'):
             emailaddrlist = body.get('notifies')
             logger.debug(emailaddrlist)
-        host = body.get('host', '')
-        if host == '':
+        # source_host is original from the message, host is remapped to CLOWDER_URL if given
+        source_host = body.get('host', '')
+        host = self.clowder_url if self.clowder_url is not None else source_host
+        if host == '' or source_host == '':
             return
-        elif not host.endswith('/'):
-            host += '/'
+        if not source_host.endswith('/'): source_host += '/'
+        if not host.endswith('/'): host += '/'
         secret_key = body.get('secretKey', '')
         retry_count = 0 if 'retry_count' not in body else body['retry_count']
         resource = self._build_resource(body, host, secret_key)
         if not resource:
             return
 
         # register extractor
-        url = "%sapi/extractors" % host
+        url = "%sapi/extractors" % source_host
```

Comment:
OK, I think I understand and there may not be any issues here - the idea of this PR is that *only* the initial downloading of the file/dataset to the extractor container uses CLOWDER_URL. It is not passed to the extractor process_message() itself, because (as discussed with Rob earlier up) a ton of extractors will use that URL to create their metadata agent. So we don't want the Clowder GUI showing "this metadata created by..." and an Amazon internal IP address URL. So pending a more comprehensive option, the CLOWDER_URL isn't passed along.

If you set CLOWDER_URL and didn't get any errors in the "Downloading file." stage it hopefully worked.